admin/attributes_controller.php

```diff
@@ -524,22 +477,22 @@
 // turned off until working
           $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
                         SET attributes_image = '" . zen_db_input($attributes_image_name) . "'
-                        WHERE products_attributes_id = " . (int)$attribute_id);
+                        WHERE products_attributes_id = " . $attribute_id);
 
           $db->Execute("UPDATE " . TABLE_PRODUCTS_ATTRIBUTES . "
-                        SET products_id = " . (int)$products_id . ",
-                            options_id = " . (int)$options_id . ",
-                            options_values_id = " . (int)$values_id . ",

```

Comment:
I'm slightly reluctant to remove these typecasts from the various sql queries, because if someone installs a 3rd party modification to earlier code that doesn't properly sanitize the query fields then it could leave things vulnerable. Plus it's faster to visually scan the code and be sure that the query is safe instead of having to go check all over to be sure.

That said, I definitely prefer `(int)`/`(float)` where possible since they're specific. And I prefer `zen_db_input` over `zen_db_prepare_input` when handling strings.

Reply:
ok I understand. I'll put the typecasts.